### ISSUE TITLE
Reaction.add_metabolites should also accept string keys

### ISSUE THREAD

**Issue by coltonlloyd** [opencobra/cobrapy#177]
```python
a = cobra.Reaction("test")
a.add_metabolites({"a": -1, "b": 1})
```

should work

**Comment by vivekiitkgp** [comment]
In the default case, the `add_metabolites` function accepts `Metabolite` objects that have several well defined attributes. In this case,  since nothing is known except the name of the metabolite, what default values do we assign to the other attributes (formula, name, compartment etc.,). We can also leave it blank, but I'm not sure if it's a good idea.

cc @aebrahim

**Comment by aebrahim** [comment]
I think this function shouldn't create metabolites (at least by default).
Instead it should look to see if a metabolite with that id exists in the
model, and raise a keyerror if not found.

If people want creation of new metabolites, the best way to do that would
be to have a create_metabolites=False keyword argument. The compartment can
be set from the id assuming the convention _c and other attributes could be
left blank.

It is important that this flag if implemented default to False, because
this behaviour isn't as well defined.

Thanks so much for taking this on

**Comment by vivekiitkgp** [comment]
Won't that be a bit confusing? Currently, the function takes metabolites dict and adds them to the Reaction and the Model. In this case, it is assumed that user already has well defined the Metabolites.

Now, when we use strings in place of objects, there can be two cases:

1. User will fill out the details of the Metabolites later. For now, just go on and add metabolites, as suggested by the name of function. One can leave a warning though ('Added an empty metabolite')
2. Specially handle the case of strings. That is, *look to see if a metabolite with that id exists in the model, and raise a keyerror if not found.* Add an additional argument.

> If people want creation of new metabolites, the best way to do that ..

I really find the current method to be the best method for adding metabolites. Create metabolites and then add them. Done! The proposition above seems non-intuitive to be. 

Probably, `add_metabolites` should just do what it does. Go and add them.

**Comment by aebrahim** [comment]
way #2 really is the way to go with this one I think. It's better to keep the "explicit is better than implicit" philosophy.

I'm going to implement the "existing Metabolite" functionality right now.

We can keep discussing how to handle id's not in the model.

**Comment by aebrahim** [comment]
(thanks for all your help thus far with cobrapy by the way. It's very appreciated)

**Comment by aebrahim** [comment]
OK, the basic feature (without creation) is now implemented in 3c1d68c

**Comment by aebrahim** [comment]
I'm marking this as closed for now, because the basic functionality has been added.

If you want metabolite creation in the add_metabolites function with a string, we can discuss that in a separate issue.
